bmap-rs/src/main.rs

```diff
@@ -89,7 +92,22 @@ fn setup_input(path: &Path) -> Result<Decoder> {
     }
 }
 
-fn copy(c: Copy) -> Result<()> {
+async fn setup_remote_input(path: &Path) -> Reqwest_Response {
+    if path.extension().unwrap() != "gz" {
+        panic!("Image file format not implemented")
```

Comment:
Should just change the return type to be `anyhow::Result` & then `bail!` instead IMO, it's a bit cleaner.